sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -3393,4 +3393,15 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
         "unsupported" -> unsupported.toString,
         "class" -> unsupported.getClass.toString))
   }
+
+  def funcBuildError(funcName: String, cause: Exception): Throwable = {
+    cause.getCause match {
+      case st: SparkThrowable with Throwable => st
```

Comment:
If some Spark's exception happens during preparation of a function call, we just propagate it to users as is otherwise (something we didn't catch) we wrap it by `AnalysisException(errorClass = FAILED_FUNCTION_CALL)`